Patch based on https://github.com/FStarLang/kremlin/pull/167
Prepared for EasyBuild by Simon Branford of the BEAR Software team at the University of Birmingham
diff -aur nss-3.51.orig/nss/lib/freebl/verified/kremlin/include/kremlin/internal/types.h nss-3.51/nss/lib/freebl/verified/kremlin/include/kremlin/internal/types.h
--- nss-3.51.orig/nss/lib/freebl/verified/kremlin/include/kremlin/internal/types.h	2020-03-23 20:23:06.943356000 +0000
+++ nss-3.51/nss/lib/freebl/verified/kremlin/include/kremlin/internal/types.h	2020-03-23 20:24:27.270377000 +0000
@@ -56,7 +56,8 @@
 #include <emmintrin.h>
 typedef __m128i FStar_UInt128_uint128;
 #elif !defined(KRML_VERIFIED_UINT128) && !defined(_MSC_VER) && \
-    (defined(__x86_64__) || defined(__x86_64) || defined(__aarch64__))
+    (defined(__x86_64__) || defined(__x86_64) || defined(__aarch64__) || \
+    (defined(__powerpc64__) && defined(__LITTLE_ENDIAN__)))
 typedef unsigned __int128 FStar_UInt128_uint128;
 #else
 typedef struct FStar_UInt128_uint128_s {
diff -aur nss-3.51.orig/nss/lib/freebl/verified/kremlin/kremlib/dist/minimal/fstar_uint128_gcc64.h nss-3.51/nss/lib/freebl/verified/kremlin/kremlib/dist/minimal/fstar_uint128_gcc64.h
--- nss-3.51.orig/nss/lib/freebl/verified/kremlin/kremlib/dist/minimal/fstar_uint128_gcc64.h	2020-03-23 20:23:06.947505000 +0000
+++ nss-3.51/nss/lib/freebl/verified/kremlin/kremlib/dist/minimal/fstar_uint128_gcc64.h	2020-03-23 20:25:20.007003000 +0000
@@ -25,7 +25,8 @@
 #include "LowStar_Endianness.h"
 
 #if !defined(KRML_VERIFIED_UINT128) && !defined(_MSC_VER) && \
-    (defined(__x86_64__) || defined(__x86_64) || defined(__aarch64__))
+    (defined(__x86_64__) || defined(__x86_64) || defined(__aarch64__) || \
+    (defined(__powerpc64__) && defined(__LITTLE_ENDIAN__)))
 
 /* GCC + using native unsigned __int128 support */
 
